src/ShapeCrawler/ShapeCollection/SlideShapes.cs

```diff
@@ -121,68 +123,66 @@ public void AddAudio(int x, int y, Stream audio, AudioType type)
     public void AddPicture(Stream image)
     {
         image.Position = 0;
-        var imageCopy = new MemoryStream();
-        image.CopyTo(imageCopy);
-        imageCopy.Position = 0;
-        image.Position = 0;
-        using var skBitmap = SKBitmap.Decode(imageCopy);
-        
-        int height;
-        int width;
-        P.Picture pPicture;
-
-        if (skBitmap != null)
+        try
         {
-            height = skBitmap.Height;
-            width = skBitmap.Width;
+            using var imageMagick = new MagickImage(
+                image,
+                new MagickReadSettings { BackgroundColor = MagickColors.Transparent });
+
+            if (imageMagick.Format == MagickFormat.Unknown)
+            {
+                throw new SCException("Unsupported image format.");
+            }
+
+            var originalFormat = imageMagick.Format;
+            if (!SupportedImageFormats.Contains(imageMagick.Format) || VectorImageFormats.Contains(imageMagick.Format))
+            {
+                imageMagick.Format = MagickFormat.Png;
```

Comment:
Should this be an option that's passed into the function? Or maybe a global setting?

Reply:
@ashahabov @jcoliz I need some input on this, and then I think this is ready.

Reply:
@blockingHD you said PowerPoint converts unsupported types to png "from what I can tell". We should be able to discern this precisely. When saving PPTX from PowerPoint, then unzipping, what is the file in the media directory?

@ashahabov In this situation, the question is: To what image file format do we convert image files of other supported formats? Once we know exactly what PowerPoint does, IMO it's fine to hard-code that.

Reply:
> To what image file format do we convert image files of other supported formats?

I've tested adding on slide five below image formats using PowerPoint:
`.jpg` -> `.jpg`
`.png` -> `.png`
`.gif` -> `.gif`
`.bmp` -> **`.png`**
`.svg` -> **`.png`** (rasterized version)

As @blockingHD has already mentioned, it looks like PowerPoint converts "unsupported" formats in `.png`. Thus, if we can, we should do the same.

Reply:
Ok, that's interesting. Subtle thing in that list. `.bmp` -> `.png`. This tells us that BMP is _not_ a "supported format", because it is converted by PowerPoint. Ergo `MagickFormat.Bmp` doesn't belong in `SupportedImageFormats` field.

This does really show the need for a test on each "supported format" to make sure it's _not_ converted.

Reply:
Hi both,

I converted an image to all possible image formats that I could using image magik then took the file types that saved from the media directory. If bmp converts to png then I may have generated a bad file or something. I'll check this out tomorrow.

Reply:
I'll also add a test for each image type added in.